packages/react/src/date-input/DateInput.tsx

```diff
@@ -48,6 +48,9 @@ export const DateInput = forwardRef<HTMLInputElement, DateInputProps>(
     const minDate = min ? new Date(min) : undefined;
 
     const handleDateChange = (date: Date) => {
+      // calendar deselects the selected date if clicked the selected date again.
+      if (!date) return;
```

Comment:
this is not the right way to achieve this - cause we could add a "reset" button to the calendar which would reset the date

better to disallow toggling date in the calendar using the `required` prop